index.bs

```diff
@@ -3448,15 +3423,19 @@ To <dfn>trigger aggregatable attribution</dfn> given an [=attribution trigger=]
 
 <h3 algorithm id="triggering-attribution">Triggering attribution</h3>
 
-To <dfn>obtain and deliver a debug report on trigger registration</dfn> given a [=trigger debug data type=] |dataType|,
-an [=attribution trigger=] |trigger| and an optional [=attribution source=]
+To <dfn>obtain and deliver a debug report on trigger registration</dfn>
+given a [=set=] of [=trigger debug data=] |dataSet|, an [=attribution trigger=] |trigger|,
+and an optional [=attribution source=]
 <dfn for="obtain and deliver a debug report on trigger registration"><var>sourceToAttribute</var></dfn>:
 
-1. Let |debugData| be the result of running [=obtain debug data on trigger registration=]
-    with |dataType|, |trigger|, |sourceToAttribute| and
-    [=obtain debug data on trigger registration/report=] set to null.
-1. If |debugData| is null, return.
-1. Run [=obtain and deliver a debug report=] with « |debugData| », |trigger|'s [=attribution trigger/reporting origin=],
+1. Let |debugDataList| be a new [=list=].
+1. [=set/iterate|For each=] |data| of |dataSet|:
+    1. Let |debugData| be the result of running [=obtain debug data on trigger registration=] with
+        |data|'s [=trigger debug data/data type=], |trigger|, |sourceToAttribute|,
+        and |data|'s [=trigger debug data/report=].
```

Comment:
Instead of passing the data type and report as separate arguments, should we just pass `data` itself?

Reply:
Passing the data type and report is more flexible if data generation is different.